### ISSUE TITLE
any chance we could expand support for more philips tv's ?

### ISSUE THREAD

**Issue by joepvanboekel** [danielperna84/ha-philipsjs#21]
we have 2 philips TV's that are recognized by HA, can get it added but then I can not control or turn on/off TV's at all

Device info
1_58PUS6504/12
by Philips
Firmware: mt5887:TPM196E_091.002.078.253

Device info
1_65PUS6121/12
by Philips
Firmware: mt5891:TPL161E_012.003.039.001

when this device is turned on, the state goes to Became unavailable, when turned of it goes to Turned off, so it seems to recognize something but not correctly

Turned off
7:20:56 PM - 4 hours ago
Became unavailable
5:53:21 PM - 5 hours ago

(I modified a custom component before so I know it can work, I believe philips2014 to work with the 65PUS6121, but my RaspberryPI crashed and I did not backup the script in time)

would it be of any help if I extract for both TV's all (needed) URL/portnumber/responses and post them back to here?

**Comment by elupus** [comment]
Yes if you post all end points that would help me a lot.

**Comment by joepvanboekel** [comment]
I think these are all I can find, although the information about jointspace is scattered everywhere around the internet
the 58PUS6504/12 does not respond (anymore?) to port 1925/1926 for some reason, so could not try commands yet, although HASS did detect the TV
(the TV is hanging outside, so I'll retry another time when i have direct access)

for the 65pus6121:
[responses.txt](https://github.com/danielperna84/ha-philipsjs/files/6313975/responses.txt)

**Comment by elupus** [comment]
Does it respond on /5/ instead of /1/?

**Comment by joepvanboekel** [comment]
> Does it respond on /5/ instead of /1/?

The 65pus only responds to /1/, the 58 had it Ports closed

how is Autodetect done? Via Some discovery protocol which also exposes firmware version or really using Port 1925/1926?

**Comment by elupus** [comment]
We manually configure api version..Then we detect if https is needed (1926) based on the system endpoint.

The annoying thing is that your data indicate some bastard mix and match between v1 and v6 api.

**Comment by joepvanboekel** [comment]
the 58PUS6504/12 seems to work OK now with the integration, I have powered it first down by removing the powercable then it started to respond to port 1925 again

regarding the 65PUS6121 could we put in some magic in the python script to add exceptions based on the firmware or something ?

**Comment by elupus** [comment]
From what i can tell the 65PUS6121 is not really sending anything odd apart from the long channel requests. could you try somethings for me?

Can you tune a single dummy channel manually so there is a single channel stored on the TV? doesn't need to have signal, just exists in the channel list.

Can you post {"powerstate": "Standby"} to the powerstate endpoint and see if it goes into standby.

**Comment by joepvanboekel** [comment]
when adding 2 channels I get response on 

GET http://192.168.39.105:1925/1/channels
{
1048704: {
"preset": "4",
"name": "-----"
},
1310848: {
"preset": "5",
"name": "-----"
}
}

GET http://192.168.39.105:1925/1/channellists
{
0: {
"name": "All TV channels",
"source": "tv"
}
}

POST http://192.168.39.105:1925/1/powerstate
{"powerstate": "Standby"}

will put TV in standby, so this works

the Philips integration also looks to start working after adding a channel on the TV, so I think the integration chokes on the fact it does not get an response from the channels.

perhaps I should not ask this here, but how and where can I add the trigger for the WOL to wake-up the TV (using the mediaplayer widget) before it was integrated in Hass we could add it using configuration.yaml but that moved away so I'm a bit confused now

**Comment by elupus** [comment]
Device trigger for "device was requested to turn on" inside gui.

**Comment by Emrvb** [comment]
@elupus 

Is there a specific reason why the wake on lan feature is not integrated in this library? I'm not really sure how the device trigger in home assistant is supposed to work. 
I can add the WOL as automation for the device trigger which seems to work, but how do I turn the TV on? Wake on Lan only wakes the TV from deep sleep.

**Comment by elupus** [comment]
> @elupus
> 
> Is there a specific reason why the wake on lan feature is not integrated in this library? I'm not really sure how the device trigger in home assistant is supposed to work.
> I can add the WOL as automation for the device trigger which seems to work, but how do I turn the TV on? Wake on Lan only wakes the TV from deep sleep.

Because it could be accomplished by automations also it can be done via IR blaster.  I did however not know that it only woke it from deep sleep.

**Comment by Emrvb** [comment]
@elupus 

Would you accept a pull request that adds native support for WOL (when a mac address is specified explicitly) and reverts to current behavior when there is no mac address set?

**Comment by elupus** [comment]
Maybe.. but fix the current behavior first.

**Comment by Emrvb** [comment]
Fair enough, but I'm unsure how that could be accomplished. I'm not familiar enough with home assistant internals. Perhaps you could elaborate on how you would fix this?
